src/main/java/de/tum/in/www1/artemis/domain/lecture/LectureUnitCompletion.java

```diff
@@ -0,0 +1,66 @@
+package de.tum.in.www1.artemis.domain.lecture;
+
+import java.io.Serial;
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+
+import javax.persistence.*;
+
+import de.tum.in.www1.artemis.domain.User;
+
+@Entity
+@Table(name = "lecture_unit_user")
+public class LectureUnitCompletion {
+
+    @EmbeddedId
+    @SuppressWarnings("unused")
+    private LectureUnitUserId id = new LectureUnitUserId();
+
+    @ManyToOne
+    @MapsId("userId")
+    private User user;
+
+    @ManyToOne
+    @MapsId("lectureUnitId")
+    private LectureUnit lectureUnit;
+
+    @Column(name = "completed_date")
+    private ZonedDateTime completedAt;
+
+    public User getUser() {
+        return user;
+    }
+
+    public void setUser(User user) {
+        this.user = user;
+    }
+
+    public LectureUnit getLectureUnit() {
+        return lectureUnit;
+    }
+
+    public void setLectureUnit(LectureUnit lectureUnit) {
+        this.lectureUnit = lectureUnit;
+    }
+
+    public ZonedDateTime getCompletedAt() {
+        return completedAt;
+    }
+
+    public void setCompletedAt(ZonedDateTime completedAt) {
+        this.completedAt = completedAt;
+    }
+
+    @Embeddable
```

Comment:
We had some issues with the plagiarism matches (and the structure) since we used `@Embeddable` - is there any chance that we can avoid this? (e.g. https://ls1tum.slack.com/archives/G6TCVL6HL/p1655130608355669)

Reply:
Hmm... I am not sure if you can transfer that discussion to here. In this class, the annotation is used to create a [composite primary key](https://www.baeldung.com/spring-jpa-embedded-method-parameters).

Reply:
Yes you are right, that should be unproblematic - but just keep an eye on it :D